### PR TITLE
fix(organizations): Fix types errors related to policies and `json.loads` function

### PR DESCRIPTION
**Pull request by puchy22**
### Context

`eu-west-1 -- TypeError[106]: the JSON object must be str, bytes or bytearray, not dict`

### Description

Add controls to ensure that in function `__describe_policy__` a dict is not passed to a `json.loads` function. 

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/organizations/organizations_service.py [hunk 1]
```diff
@@ -140,14 +140,18 @@ def __describe_policy__(self, policy_id):
                     .get("Policy", {})
                     .get("Content", "")
                 )
+
+                if isinstance(policy_content, str):
+                    policy_content = json.loads(policy_content)
```
Comment (jfagoagas) [inline prowler/providers/aws/services/organizations/organizations_service.py:145]:
```suggestion
                if isinstance(policy_content, str):
                    policy_content = json.loads(policy_content)
                return policy_content
```

File: prowler/providers/aws/services/organizations/organizations_service.py [hunk 2]
```diff
@@ -140,14 +140,18 @@ def __describe_policy__(self, policy_id):
...
+
         except Exception as error:
             logger.error(
                 f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
```
Comment (jfagoagas) [inline prowler/providers/aws/services/organizations/organizations_service.py:151]:
```suggestion
            )
            return {}
```

File: prowler/providers/aws/services/organizations/organizations_service.py [hunk 3]
```diff
@@ -140,14 +140,18 @@ def __describe_policy__(self, policy_id):
...
         finally:
-            return json.loads(policy_content)
+            return policy_content if isinstance(policy_content, dict) else {}
```
Comment (jfagoagas) [inline prowler/providers/aws/services/organizations/organizations_service.py]: I'd remove the finally block entirely.

File: prowler/providers/aws/services/organizations/organizations_service.py [hunk 4]
```diff
@@ -161,7 +165,7 @@ def __list_targets_for_policy__(self, policy_id):
                 f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
         finally:
-            return targets_for_policy
+            return targets_for_policy if isinstance(targets_for_policy, list) else []
```
Comment (jfagoagas) [inline prowler/providers/aws/services/organizations/organizations_service.py]: The same here as above.

**Review by jfagoagas — DISMISSED** [review]
I think the PR body could lead into confusion. The issue here is that if `self.client.describe_policy` fails we are trying to do `json.loads()` with a `dict` object since we create `policy_content = {}` at the beginning of the function.

Please review my comments when you get a chance. Thanks!

File: prowler/providers/aws/services/organizations/organizations_service.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/organizations/organizations_service.py:151]: Please, add this

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**